src/frontend/src/lib/components/carousel/Carousel.svelte

```diff
@@ -0,0 +1,235 @@
+<script lang="ts">
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { onMount } from 'svelte';
+	import Controls from '$lib/components/carousel/Controls.svelte';
+	import Indicators from '$lib/components/carousel/Indicators.svelte';
+	import {
+		updateSliderFrameVisiblePart,
+		buildCarouselSliderFrame,
+		disableTransition,
+		enableTransition
+	} from '$lib/utils/carousel.utils';
+
+	export let autoplay = 5000;
+	export let duration = 300;
+	export let easing = 'ease-out';
+	export let styleClass: string | undefined = undefined;
+
+	/**
+	 * Carousel container element variables
+	 */
+	let container: HTMLDivElement | undefined;
+	let containerWidth = 0;
+
+	/**
+	 * Computed slider frame element that wraps all slides
+	 */
+	let sliderFrame: HTMLDivElement | undefined;
+
+	/**
+	 * Variables related to the slides
+	 */
+	let slides: Node[];
+	let currentSlide = 0;
+	let totalSlides: number;
+
+	/**
+	 * Autoplay timer
+	 */
+	let autoplayTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * Slide transition timer that needed for last-to-first and first-to-last non-animated transform
+	 */
+	let slideTransformTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * A const to be used for enabling transition animation
+	 */
+	const animationParams = {
+		easing,
+		duration
+	};
+
+	/**
+	 * Initialise all required data and attach/detach events
+	 */
+	onMount(() => {
+		initializeCarousel();
+		initialiseAutoplayTimer();
+
+		// Resize listener to re-calculate slide frame width
+		window.addEventListener('resize', onResize);
+
+		return () => {
+			clearAutoplayTimer();
+			clearSlideTransformTimer();
+
+			window.removeEventListener('resize', onResize);
+		};
+	});
+
+	/**
+	 * Build slider frame and set required variables
+	 */
+	const initializeCarousel = () => {
+		if (isNullish(container)) {
+			return;
+		}
+
+		// Save this data only on mount
+		if (isNullish(slides)) {
+			slides = [...container.children];
+			totalSlides = slides.length;
+		}
+
+		containerWidth = container.offsetWidth;
+
+		sliderFrame = buildCarouselSliderFrame({
+			slides,
+			slideWidth: containerWidth
+		});
+
+		// Clean previous HTML if the frame is being re-built (e.g. in case of window resize)
+		container.innerHTML = '';
+		container.appendChild(sliderFrame);
+
+		goToSlide(currentSlide);
+	};
```

Comment:
Many things need to be initialized, I'm a bit afraid we end up in a partial state in which some information are set and some not. Should we prevent those kind of desynchronization?

Reply:
I added a small check to see if sliderFrame includes slides, couldn't come up with any other solutions here

Reply:
Oh that's another thing that needs to be documented - if a slide gets removed dynamically after everything is mounted then I guess the all slider is broken?

Reply:
I don't think it can be the case, I added that check mostly to attempt to prevent the above mentioned desynchronization. If it looks confusing, I can remove it.

Reply:
What I mean is, if I understand correctly, the carousel will work fine as long as there are no dynamic changes to its content, like adding or removing slides. I'm also not sure how it would behave if one of the slides is a Svelte component that gets rendered while the carousel is mostly a static item.

Long story short, I foresee quite a potential for errors if this component is reused in any dynamic environment.

It's not a blocker for this PR, but definitely something to tackle once the due date has passed.

Reply:
I will keep it in mind while implementing the next step where I actually use the component. I think it won't be a big deal to adjust some values to be dynamic, but in general your point totally makes sense.